homeassistant/components/nextbus/config_flow.py

```diff
@@ -0,0 +1,212 @@
+
+    async def async_step_agency(
+        self,
+        user_input: UserConfig | None = None,
+    ) -> FlowResult:
+        """Select agency."""
+        if user_input is not None:
+            self.data[CONF_AGENCY] = user_input[CONF_AGENCY]
+
+            return await self.async_step_route()
+
+        self._agency_tags = await self.hass.async_add_executor_job(
+            _get_agency_tags, self._client
+        )
+
+        return self.async_show_form(
+            step_id="agency",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_AGENCY): _dict_to_select_selector(
+                        self._agency_tags
+                    ),
+                }
+            ),
+        )
+
+    async def async_step_route(
+        self,
+        user_input: UserConfig | None = None,
+    ) -> FlowResult:
+        """Select route."""
+        if user_input is not None:
+            self.data[CONF_ROUTE] = user_input[CONF_ROUTE]
+
+            return await self.async_step_stop()
+
+        self._route_tags = await self.hass.async_add_executor_job(
+            _get_route_tags, self._client, self.data[CONF_AGENCY]
+        )
+
+        return self.async_show_form(
+            step_id="route",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_ROUTE): _dict_to_select_selector(
+                        self._route_tags
+                    ),
+                }
+            ),
+        )
+
+    async def async_step_stop(
+        self,
+        user_input: UserConfig | None = None,
+    ) -> FlowResult:
+        """Select stop."""
+
+        if user_input is not None:
+            self.data[CONF_STOP] = user_input[CONF_STOP]
+
+            return await self.async_step_name()
+
+        self._stop_tags = await self.hass.async_add_executor_job(
+            _get_stop_tags,
+            self._client,
+            self.data[CONF_AGENCY],
+            self.data[CONF_ROUTE],
+        )
+
+        return self.async_show_form(
+            step_id="stop",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_STOP): _dict_to_select_selector(self._stop_tags),
+                }
+            ),
+        )
+
+    async def async_step_name(
+        self,
+        _: UserConfig | None = None,
+    ) -> FlowResult:
+        """Set name based on agency, route, and stop."""
+
+        # Abort if duplicate entries exist
+        self._async_abort_entries_match(self.data)
+
+        agency_tag = self.data[CONF_AGENCY]
+        route_tag = self.data[CONF_ROUTE]
+        stop_tag = self.data[CONF_STOP]
+
+        agency_name = self._agency_tags[agency_tag]
+        route_name = self._route_tags[route_tag]
+        stop_name = self._stop_tags[stop_tag]
+
+        return self.async_create_entry(
+            title=f"{agency_name} {route_name} {stop_name}",
+            data=self.data,
+        )
```

Comment:
Please delete `async_step_name` function, and move the content inside `async_step_stop`